assets/app/scripts/controllers/logs/buildLog.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('BuildLog', [
+    '$anchorScroll',
+    '$location',
+    '$q',
+    '$routeParams',
+    '$scope',
+    '$timeout',
+    '$window',
+    'DataService',
+    'project',
+    'logLinks',
+    function($anchorScroll, $location, $q, $routeParams, $scope, $timeout, $window, DataService, project, logLinks) {
+
+      DataService.list("projects", $scope, function(projects) {
+        $scope.projects = projects.by("metadata.name");
+        Logger.log("projects", $scope.projects);
+      });
+
+      project
+        .get($routeParams.project)
+        .then(_.spread(function(project, context) {
+          return $q.all([
+                    DataService
+                      .get('builds', $routeParams.build, context),
+                    DataService
+                      .get('builds/log', $routeParams.build, context)
+                  ])
+                  .then(_.spread(function(build, log) {
```

Comment:
Still thinking about this.  The UI would not be unresponsive since the calls are async.  A loading spinner is a pretty standard interface for this (I have one in logs), and other page controls will work fine.  I really like the elegance that `$q.all()` provides, especially with `.then(_.spread(function(first, second) { }))`.  It improves readability & I'd argue maintainability.

There is definitely good reason to isolate promise calls, but I tend to think we should use `$q.all().then(_.spread())` as the primary idiom and only separate the async calls when the experience proves to be poor & needs an extra level of optimization.